sdk/core/azure-core/src/main/java/com/azure/android/core/http/interceptors/UserAgentInterceptor.java

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.core.http.interceptors;
+
+import androidx.annotation.NonNull;
+
+import com.azure.android.core.util.Configuration;
+import com.azure.android.core.util.ServiceVersion;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * Pipeline interceptor that adds "User-Agent" header to a request.
+ * <p>
+ * The format for the "User-Agent" string is outlined in
+ * <a href="https://azure.github.io/azure-sdk/general_azurecore.html#telemetry-policy">Azure Core: Telemetry policy</a>.
+ */
+public class UserAgentInterceptor implements Interceptor {
+    private static final String DEFAULT_USER_AGENT_HEADER = "azsdk-java";
```

Comment:
Probably needs to change, my guess is `azsdk-android` but I would check with Adrian.